pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -680,6 +680,20 @@ def test_boolean_mask_nullable_int64(self):
         )
         tm.assert_frame_equal(result, expected)
 
+    def test_setitem_npmatrix_2d(self):
+        # for use-case df["x"] = sparse.random(10, 10).mean(axis=1)
+        expected = DataFrame(
+            {"np-array": np.ones(10), "np-matrix": np.ones(10)}, index=np.arange(10)
+        )
+
+        a = np.ones((10, 1))
+        df = DataFrame(index=np.arange(10))
+        df["np-array"] = a
+        with tm.assert_produces_warning(PendingDeprecationWarning):
```

Comment:
can you add a comment of why there is a deprecation warning

Reply:
:+1: